.github/workflows/coverage.yaml

```diff
@@ -0,0 +1,24 @@
+name: coverage
+on:
+  pull_request:
+    branches:
+    - main
+  push:
+    branches:
+    - main
+jobs:
+  run_unit_tests:
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - run: docker-compose -f ./docker-compose.yml up --detach redis
+    - uses: actions/setup-python@v3
+      with:
+        python-version: "3.10.x"
+    - run: source ./activate.sh
+    - run: pip install --upgrade pip
+    - run: pip install -e .
+    - run: pip freeze
+    - run: ./test.sh
+    - uses: codecov/codecov-action@v3.1.0
```

Comment:
Also need to add `CODECOV_TOKEN` in Github secret

Reply:
Is it this token? It says "Token not required for public repositories uploading from [Travis](https://travis-ci.org/), [CircleCI](https://circleci.com/), [AppVeyor](https://www.appveyor.com/), [Azure Pipelines](https://azure.microsoft.com/en-us/services/devops/pipelines/) or [GitHub Actions](https://help.github.com/en/actions)" so I'm wondering if we I really need to add the token.
![Screenshot from 2022-06-19 07-00-43](https://user-images.githubusercontent.com/5580715/174477829-f9c16511-f2a6-42c5-9efc-b13cedcb7961.png)

Reply:
Ok, not really sure as well but I guess no token needed for this